app/src/main/java/org/onionshare/android/server/WebserverManager.kt

```diff
@@ -0,0 +1,116 @@
+package org.onionshare.android.server
+
+import android.util.Base64
+import android.util.Base64.NO_PADDING
+import android.util.Base64.URL_SAFE
+import com.mitchellbosecke.pebble.loader.ClasspathLoader
+import io.ktor.application.Application
+import io.ktor.application.ApplicationStarted
+import io.ktor.application.ApplicationStopped
+import io.ktor.application.call
+import io.ktor.application.install
+import io.ktor.features.CallLogging
+import io.ktor.features.StatusPages
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.content.resources
+import io.ktor.http.content.static
+import io.ktor.pebble.Pebble
+import io.ktor.pebble.PebbleContent
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import io.ktor.routing.routing
+import io.ktor.server.engine.ApplicationEngine
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.netty.Netty
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import java.security.SecureRandom
+import javax.inject.Inject
+
+internal const val PORT: Int = 17638
+
+class WebserverManager @Inject constructor() {
+
+    enum class State { STARTING, STARTED, STOPPING, STOPPED }
+
+    private val _state = MutableStateFlow(State.STOPPED)
+    val state: StateFlow<State> = _state
```

Comment:
This is a strange, but common pattern to not expose the mutable nature to the outside.